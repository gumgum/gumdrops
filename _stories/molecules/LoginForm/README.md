**NOTE:** The minimum DS version required for this component to render properly is: **1.3.5**

The `<LoginForm>`, as the name implies, renders a form that can be customized to look like any of [these prototypes](http://ds.gumgum.com/stable/#gds-account-modal).
This component can display the Product name (underneath GumGum's logo), a Cap with text above the form fields and a link for password recovery. If you do not want to show GumGum's logo, you can pass the prop `hideLogo`. The form inputs and submit buttons must be provided.

All of its atoms are optional.

Form values can be accessed on submission with the `onSubmit` callback.

**Props**:

prop name   | description
------------|------------
capText         | Text to display in the cap between the logo and form.
logoText        | Product name, shown right below GumGum's logo
hideLogo        | Boolean indicating whether or not you would like to hide GumGum's logo
onSubmit        | (Required) function called on form submition
recoveryFn      | Callback used to redirect to a password recovery page
recoveryText    | Text to show for password recovery, defauts to "Forgot your password?"

**Example**:
```
state {
    username: null,
    password: null,
    staySignedIn: false,
}

_login = (event) => {
    event.preventDefault();
    const elts = event.target.elements;
    this.setState({
        username: elts['username'].value,
        password: elts['password'].value,
        staySignedIn: elts['password'].checked
    });
}

_recovery = (event) => {
    event.preventDefault();
    window.location.pathname = '/recovery';
}

return(
    <LayoutContainer>
        <Row>
            <LoginForm
                auth="username"
                capText="Dashboard"
                logoText="Sports"
                onSubmit={ this._login }
                recoveryFn={ this._goToRecovery }
                recoveryText="Trouble signing in?"
            >
                <FormGroup className="-m-b-1" >
                    <FormGroupLabel text="username" />
                    <TextInput name="username" type="text" placeholder="username" />
                </FormGroup>
                <FormGroup className="-m-b-2" >
                    <FormGroupLabel text="Password" />
                    <TextInput name="password" type="password" placeholder="password123" />
                </FormGroup>
                <FormGroup className="-m-b-2" >
                    <Checkbox name="staySignedIn" label="Stay signed in" />
                </FormGroup>
                <Button type="submit" className="gds-button--block" context="primary" >
                    Login
                </Button>
            </LoginForm>
        </Row>
    </LayoutContainer>
)
```
