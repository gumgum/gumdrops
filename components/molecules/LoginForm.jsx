import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const wrapperStyle = { position: 'relative' };

const LoginForm = ({
    capText,
    children,
    hideLogo,
    logoText,
    onSubmit,
    recoveryFn,
    recoveryText
}) => (
    <div
        className={cx('gds-account-modal', { 'gds-account-modal--logo': !hideLogo })}
        style={wrapperStyle}>
        <div className="gds-account-modal__form gds-card -p-t-3-sm -p-t-0">
            {logoText && <div className="gds-account-modal__logo-product">{logoText}</div>}
            {capText && (
                <div className="gds-account-modal__form-cap -m-t-3-sm -m-t-0">{capText}</div>
            )}
            <form className="gds-form -pointer-events--auto -p-a-3" onSubmit={onSubmit}>
                {children}
            </form>
        </div>
        {recoveryFn && (
            <p className="gds-account-modal__extra-links">
                <a onClick={recoveryFn} className="gds-text--link -cursor--pointer">
                    {recoveryText}
                </a>
            </p>
        )}
    </div>
);

LoginForm.displayName = 'LoginForm';

LoginForm.defaultProps = {
    capText: '',
    logoText: '',
    recoveryText: 'Forgot your password?',
    hideLogo: false
};

LoginForm.propTypes = {
    capText: PropTypes.string,
    children: PropTypes.node.isRequired,
    logoText: PropTypes.string,
    onSubmit: PropTypes.func.isRequired,
    recoveryFn: PropTypes.func,
    recoveryText: PropTypes.string,
    hideLogo: PropTypes.bool
};

export default LoginForm;
