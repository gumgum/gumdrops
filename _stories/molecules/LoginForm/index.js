import React from 'react';
import { boolean, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import readme from './README.md';

import Checkbox from '../../../components/molecules/Checkbox';
import LoginForm from '../../../components/molecules/LoginForm';
import FormGroup from '../../../components/molecules/FormGroup';
import FormGroupLabel from '../../../components/atoms/FormGroupLabel';
import Button from '../../../components/atoms/Button';
import TextInput from '../../../components/atoms/TextInput';

const component = () => (
    <LoginForm
        capText={text('Cap Text', 'Welcome')}
        logoText={text('Logo Text', 'Storybook')}
        hideLogo={boolean('Hide Logo', false)}
        onSubmit={action('Form Submit')}
        recoveryFn={action('Recovery Action')}
        recoveryText={text('Recovery Text', 'Forgot Password?')}>
        <FormGroup className="-m-b-1">
            <FormGroupLabel text="username" />
            <TextInput name="username" type="text" placeholder="username" />
        </FormGroup>
        <FormGroup className="-m-b-2">
            <FormGroupLabel text="Password" />
            <TextInput name="password" type="password" placeholder="password123" />
        </FormGroup>
        <FormGroup className="-m-b-2">
            <Checkbox name="staySignedIn" label="Stay signed in" />
        </FormGroup>
        <Button type="submit" className="gds-button--block" context="primary">
            Login
        </Button>
    </LoginForm>
);

export default [readme, component];
