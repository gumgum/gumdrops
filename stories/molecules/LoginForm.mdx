import { Canvas, Source } from '@storybook/blocks';
import { ArgsTable } from '@storybook/addon-docs';

import * as LoginFormStories from './LoginForm.stories';
import LoginForm from '../../components/molecules/LoginForm';

# LoginForm

**NOTE:** The minimum DS version required for this component to render properly is: **1.3.5**

The `<LoginForm>`, as the name implies, renders a form that can be customized to be used as a login form.
This component can display the Product name (underneath GumGum's logo), a Cap with text above the form fields and a link for password recovery. If you do not want to show GumGum's logo, you can pass the prop `hideLogo`. The form inputs and submit buttons must be provided.

To show a link to recover your password, you must pass in `recoveryFn` and `recoveryText`.

All of its atoms are optional.

Form values can be accessed on submission with the `onSubmit` callback.

## Full Example

```jsx
const [form, setForm] = useState({
    staySignedIn: false,
    username: '',
    password: ''
});
const _login = event => {
    event.preventDefault();
    // Do something with form
    console.log(form);
};

const _recovery = event => {
    event.preventDefault();
    window.location.pathname = '/recovery';
};

return (
    <div>
        {' '}
        <LoginForm
            auth="username"
            capText="Dashboard"
            logoText="GumGum"
            onSubmit={_login}
            recoveryFn={_recovery}
            recoveryText="Trouble signing in?">
            <FormGroup className="-m-b-1">
                <FormGroupLabel text="username" />
                <TextInput
                    onChange={e => {
                        setForm({
                            ...form,
                            username: e.target.value
                        });
                    }}
                    value={form.username}
                    name="username"
                    type="text"
                    placeholder="username"
                />
            </FormGroup>
            <FormGroup className="-m-b-2">
                <FormGroupLabel text="Password" />
                <TextInput
                    onChange={e => {
                        setForm({
                            ...form,
                            password: e.target.value
                        });
                    }}
                    name="password"
                    value={form.password}
                    type="password"
                    placeholder="password123"
                />
            </FormGroup>
            <FormGroup className="-m-b-2">
                <Checkbox
                    onChange={e => {
                        setForm({
                            ...form,
                            staySignedIn: e.target.value
                        });
                    }}
                    value={form.staySignedIn}
                    name="staySignedIn"
                    label="Stay signed in"
                />
            </FormGroup>
            <Button type="submit" className="gds-button--block" context="primary">
                Login
            </Button>
        </LoginForm>
    </div>
);
```

## Component Example

<Canvas of={LoginFormStories.Default} />

## Component Example Source

<Source of={LoginFormStories.Default} />

## LoginForm Props

<ArgsTable of={LoginForm} />
