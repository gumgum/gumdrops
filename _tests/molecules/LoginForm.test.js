import renderer from 'react-test-renderer';
import React from 'react';
import LoginForm from '../../components/molecules/LoginForm';

const defaultProps = {
    capText: 'Text',
    logoText: 'Logo Text',
    onSubmit: jest.fn(),
    recoveryFn: jest.fn(),
    recoveryText: 'Recovery message'
};

test('Expect <LoginForm> to render properly', () => {
    const tree = renderer
        .create(
            <LoginForm {...defaultProps}>
                <input type="text" />
            </LoginForm>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
