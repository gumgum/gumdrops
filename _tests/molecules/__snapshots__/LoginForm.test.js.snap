// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <LoginForm> to render 1`] = `
<LoginForm
  capText="Text"
  hideLogo={false}
  logoText="Logo Text"
  onSubmit={[Function]}
  recoveryFn={[Function]}
  recoveryText="Recovery message"
>
  <div
    className="gds-account-modal gds-account-modal--logo"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="gds-account-modal__form gds-card -p-t-3-sm -p-t-0"
    >
      <div
        className="gds-account-modal__logo-product"
      >
        Logo Text
      </div>
      <div
        className="gds-account-modal__form-cap -m-t-3-sm -m-t-0"
      >
        Text
      </div>
      <form
        className="gds-form -pointer-events--auto -p-a-3"
        onSubmit={[Function]}
      >
        <input
          type="text"
        />
      </form>
    </div>
    <p
      className="gds-account-modal__extra-links"
    >
      <a
        className="gds-text--link -cursor--pointer"
        onClick={[Function]}
      >
        Recovery message
      </a>
    </p>
  </div>
</LoginForm>
`;
